src/app/services/time/time.service.ts

```diff
@@ -1,15 +1,17 @@
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
-import { ITime } from '../../time';
+import { ITime } from '../../models/time';
 
-@Injectable({
-  providedIn: 'root'
-})
+@Injectable()
 export class TimeService {
 
   constructor(private http: HttpClient) { }
 
+  private url: string = "https://data-visualization-back-end.herokuapp.com";
```

Comment:
Minor point: this constant can be shared between services to avoid duplication.

Reply:
Best practice would also usually to externalise this URL as config somehow as per https://12factor.net/config

This isn't required now when you only the one environment but would be once you add any more (or if you want to run both the front end an back end locally).

Not sure if this is in the scope of the project? Probably a question for @martinrwilson?

If so you could add it as sub task on a story  to create staging/production pipeline (for example).

Reply:
It probably is worth doing this, as it's best practice and probably not too much effort ..(?)

Reply:
For Javascript apps & build pipelines it's actually a bit more difficult than it should be.

I think there's a way in Angular which isn't fully 12 factor (i.e. build once deploy anywhere) but will allow you to have separate config files for know environments (e.g. dev, test, staging, and production)

I'd push it back for now and when it's required one of us can give him a hand.